apps/system/js/base_module.js

```diff
@@ -110,7 +102,83 @@
    */
   BaseModule.STATES = [];
 
+  BaseModule.SIDE_MODULES = [];
```

Comment:
Also I'm not sure that we need a top level `SIDE_MODULES` concept if each module is responsible to trigger the `_startSideModules()`.

Either the logic of "when should we load side modules" is something common, then we should have it in the "declarative part" of the `BaseModule` (ie. `BaseModule.SIDE_MODULES`) and the default implementation should take care of loading them.
Or the load logic really depends on each module and we should just add a `BaseModule.lazyLoad('ModuleName')` helper method and let each module explicitly load stuff lazily when it fits.